lib/Sema/CSSimplify.cpp

```diff
@@ -7097,8 +7097,37 @@ ConstraintSystem::simplifyApplicableFnConstraint(
   }
 
   // Handle applications of @dynamicCallable types.
-  return simplifyDynamicCallableApplicableFnConstraint(type1, origType2,
-                                                       subflags, locator);
+  auto result = simplifyDynamicCallableApplicableFnConstraint(
+      type1, origType2, subflags, locator);
+
+  if (shouldAttemptFixes() && result == SolutionKind::Error) {
+    // Skip this fix if the type is not yet resolved or
+    // it's a function type/metatype which points to argument mismatches.
+    if (desugar2->is<TypeVariableType>() || desugar2->is<FunctionType>() ||
+        desugar2->is<AnyMetatypeType>())
+      return SolutionKind::Error;
+
+    if (auto objectTy = desugar2->lookThroughAllOptionalTypes()) {
+      if (objectTy->isAny() || objectTy->isAnyObject())
+        return SolutionKind::Error;
+    }
+
+    // If there are any type variables associated with arguments/result
+    // they have to be marked as "holes".
+    type1.visit([&](Type subType) {
+      if (auto *typeVar = subType->getAs<TypeVariableType>())
+        recordHole(typeVar);
+    });
+
+    auto *fix = RemoveInvalidCall::create(*this, getConstraintLocator(locator));
+    // Let's make this fix as high impact so if there is a function or member
+    // overload with e.g. argument-to-parameter type mismatches it would take
+    // a higher priority.
+    return recordFix(fix, /*impact=*/10) ? SolutionKind::Error
```

Comment:
This isn't specific to this change or something that needs to be addressed now, but what do you think about making `impact` a method on `ConstraintFix` that subclasses could override as needed? I think it would make it easier to determine the relative importance of fixes by just scanning the `CSFix` header.

Reply:
Oops, didn't see @hborla's comment before posting

Reply:
I was thinking about doing that as well, but some far most of the impact changes where contextual in nature. Although maybe that's a nice long-hanging fruit to tackle for clarity...

Reply:
My hope is once we finish porting majority of diagnostics we would become more clear whether we can stick to weight or need something more comprehensive than that...